src/main/java/com/google/research/bleth/simulator/SimulationMetadata.java

```diff
@@ -36,8 +36,8 @@ public SimulationMetadata(AbstractSimulation.Builder builder) {
         this.observersNum = builder.getObserversNum();
         this.rowsNum = builder.getRowNum();
         this.colsNum = builder.getColNum();
-        this.beaconMovementStrategy = builder.getBeaconMovementStrategy().toString();
-        this.observerMovementStrategy = builder.getObserverMovementStrategy().toString();
+        this.beaconMovementStrategy = builder.getBeaconMovementStrategy().getClass().toString();
```

Comment:
Does the builder hold a specific instance - i.e. the movement strategy of a specific agent?
If we can create new instance of the strategy from a string - why not store that string?

Reply:
Actually that's a good point. Right now the Simulation Builder gets an instance of a movement strategy, and initialize all beacons/observers with that instance. We actually want each agent have its own movement strategy object (even if they are all instances of the same class). 

To solve this we can use the same solution we used to create awakeness strategies (the builder gets the strategy type, and the strategies instances created by a factory).

Reply:
Sounds good.